src/json/json.cc

```diff
@@ -0,0 +1,3232 @@
+    uint64_t val = ValkeyModule_LoadUnsigned(ctx);
+    if (value != val) {
+        ValkeyModule_Log(nullptr, "warning", "%s: Unexpected value in RDB Expected: %lx Received: %lx",
+                        caller, value, val);
+        return false;
+    }
+    return true;
+}
+
+/*
+ * Check the encoding version, For unsupported versions we ALWAYS put out a message in the log
+ * but we only fail the RDB load if the config tells us to do it.
+ */
+bool checkVersion(const char *type_name, int encver, int expected_encver) {
+    if (encver != expected_encver) {
+        if (enforce_rdb_version_check) {
+            ValkeyModule_Log(nullptr, "warning", "Unsupported Encoding Version %d for type:%s expected %d",
+                            encver, type_name, expected_encver);
+            return false;
+        } else {
+            ValkeyModule_Log(nullptr, "warning",
+                            "Unsupported Encoding Version %d for type:%s expected %d, WILL ATTEMPT LOADING ANYWAYS",
+                            encver, type_name, expected_encver);
+        }
+    }
+    return true;
+}
+
+bool checkVersionRange(const char *type_name, int encver, int ver_low, int ver_high) {
+    if (encver < ver_low || encver > ver_high) {
+        if (enforce_rdb_version_check) {
+            ValkeyModule_Log(nullptr, "warning", "Unsupported Encoding Version %d for type:%s expected [%d:%d]",
+                            encver, type_name, ver_low, ver_high);
+            return false;
+        } else {
+            ValkeyModule_Log(nullptr, "warning",
+                            "Unsupported Encoding Version %d for type:%s expected [%d:%d],"
+                            " WILL ATTEMPT TO LOAD ANYWAYS",
+                            encver, type_name, ver_low, ver_high);
+        }
+    }
+    return true;
+}
+
+/*
+ * Stub for scdtype0 data type.
+ */
+#define SCDTYPE_ENCVER 1
+int scdtype_aux_load(ValkeyModuleIO *ctx, int encver, int when) {
+    if (!checkVersion("sdctype0", encver, SCDTYPE_ENCVER)) return VALKEYMODULE_ERR;
+    if (when == VALKEYMODULE_AUX_AFTER_RDB) {
+        if (!checkInt(ctx, 0, "scdtype")) return VALKEYMODULE_ERR;
+    }
+    return VALKEYMODULE_OK;
+}
+
+#define GEARSDT_ENCVER 3
+int gearsdt_aux_load(ValkeyModuleIO *ctx, int encver, int when) {
+    if (!checkVersion("gearsdt", encver, GEARSDT_ENCVER)) return VALKEYMODULE_ERR;
+    if (when == VALKEYMODULE_AUX_AFTER_RDB) {
+        if (!checkString(ctx, "StreamReader", "gears-dt")) return VALKEYMODULE_ERR;
+        if (!checkInt(ctx, 0, "gears-dt")) return VALKEYMODULE_ERR;
+        if (!checkString(ctx, "CommandReader", "gears-dt")) return VALKEYMODULE_ERR;
+        if (!checkInt(ctx, 0, "gears-dt")) return VALKEYMODULE_ERR;
+        if (!checkString(ctx, "KeysReader", "gears-dt")) return VALKEYMODULE_ERR;
+        if (!checkInt(ctx, 0, "gears-dt")) return VALKEYMODULE_ERR;
+        if (!checkString(ctx, "", "gears-dt")) return VALKEYMODULE_ERR;
+    }
```

Comment:
these checks add fragility only to the module compat. Since we are not planning to consume any of them, the best strategy imo would be to log a "xxx is skipped" notice level message and return _OK. same goes to other stubs.

Reply:
> yeah these aux data types seem to just play a supporting role so far. can we register a generic stub instead and just log a message that "xxx is skipped" and move on? there could be more aux types added in the future in rejson so it will be good to have that visibility if community members report any issue.

The challenge is that each aux type may have different number of entries in RDB. For example, scdtype only has one integer, while gearsdt has 7 entries in RDB (some are int and some are string).

Reply:
Got it - the parsing part makes sense to me now. But maybe we could skip the check of the actual value?

Reply:
Totally agreed, no need to check the actual value.

Reply:
https://github.com/valkey-io/valkeyJSON/issues/8